### DISCUSSION TITLE
Store Email parameters and use PHP Include to add them across multiple pages

### DISCUSSION THREAD

**Discussion by EnjoySinglemalt** [PHPMailer/PHPMailer#2866] [Q&A]
Hey all, I have a need to store the Email server connection parameters in a separate file so I can pull them into multiple scripts. I tried to use the PHP function 'Include' to add these, but I could not get this to work. I also tried 'Require' but no combination worked. 
Sample code:

```php
<?php
echo 'Starting Email Script<br>';
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
date_default_timezone_set('Etc/UTC');
require '../vendor/autoload.php';

$mail = new PHPMailer(); // the true param means it will throw exceptions on errors, which we need to catch
$mail->isSMTP(); // telling the class to use SMTP
//Bring in Connection Paramaters
include 'email_connection.php'; //Try #1
require_once '/var/www/acts/email_connection.php'; //Try #2
include $_SERVER['DOCUMENT_ROOT'].'email_connection.php'; //Try #3
......
_Rest of email messages parameters below_
?>
```

**Comment by Synchro** [comment]
You show loading the files, but not what's in them or how you're trying to use the values set in them. Make sure you read [the docs on `include`](https://www.php.net/manual/en/function.include.php) and how its scopes work.

I would recommend using a library specifically for doing this, such as [phpdotenv](https://github.com/vlucas/phpdotenv)

**Reply by EnjoySinglemalt** [reply]
Thanks for the reply. Here are the parameters I am trying to pull in:

```
<?php

  //Connection Paramaters	
  $mail->SMTPDebug = SMTP::DEBUG_SERVER; // enables SMTP debug information (for testing)
  $mail->Host = "smtp.example.comt"; // sets the SMTP server
  $mail->Port = 587; // set the SMTP port 
  $mail->SMTPAuth = true; // enable SMTP authentication
  $mail->Username = "emailusername"; // SMTP account username
  $mail->Password = "emailpassword"; // SMTP account password
?>
```

If I add these parameters into a test email script I can successfully send emails.

**Reply by Synchro** [reply]
This approach relies on `$mail` existing and being in scope when this is called, which is quite likely to break. I'd suggest doing it like this:

```php
$debug = SMTP::DEBUG_SERVER;
$host = 'smtp.example.com';
$port = 587;
etc
```
then in the file you include it into:

```php
$mail->SMTPDebug = $debug;
$mail->Host = $host;
$mail->post = $port;
etc
```

Alternatively, put all of these things into a single array so that it's more obviously coming from config:

```php
$config = [
  'debug' => SMTP::DEBUG_SERVER,
  'host' => 'smtp.example.com',
  'port' => 587
];
```

but now this is getting to be pretty much what phpdotenv does for you, but it's much more capable.

**Reply by EnjoySinglemalt** [reply]
Marcus, thanks for the prompt response and the suggestions. I am a very part-time coder so I am always looking for better ways to handle use cases. I am looking at the documentation for phpdotenv and this seems like a good direction to go.
